lib/bsdlib/nrf91_sockets.c

```diff
@@ -692,6 +726,17 @@ static int nrf91_socket_offload_getaddrinfo(const char *node,
 		} else if (error == -EAFNOSUPPORT) {
 			return DNS_EAI_ADDRFAMILY;
 		}
+		if (hints->ai_next != NULL)
+		{
+			nrf_hints_pdn_ptr = &nrf_hints_pdn;
+			z_to_nrf_addrinfo_hints(hints->ai_next, &nrf_hints_pdn);
+			if (error == -EPROTONOSUPPORT) {
+				return DNS_EAI_SOCKTYPE;
+			} else if (error == -EAFNOSUPPORT) {
+				return DNS_EAI_ADDRFAMILY;
+			}
+			nrf_hints_ptr->ai_next = nrf_hints_pdn_ptr;
```

Comment:
Is `nrf_hints_pdn_ptr` really needed? You could just assign `&nrf_hints_pdn` directly here.

Reply:
i just follewed the same style as for nrf_hints and nrf_hints_ptr. I could probably refactor the old code as well, to remove the *ptr for hints passing in the struct directly.

Reply:
Well, the majority of code here was merged in bulk and in a pre-release rush, so some peculiarities like this might've been merged unnoticed. So if there's a chance to improve now, we should go after it, not duplicate the issues.

Reply:
Good point! :) 
I have amended the commit to remove both the nrf_hints pointers.